src/register.jl

```diff
@@ -326,8 +326,25 @@ end
 function update_versions_file(pkg::Project,
                               versions_file::AbstractString,
                               versions_data::Dict{String, Any},
-                              tree_hash::AbstractString)
+                              tree_hash::AbstractString;
+                              commit_hash::Union{AbstractString,Nothing}=nothing,
+                              tag_hash::Union{AbstractString,Nothing}=nothing,
+                              tag_name::Union{AbstractString,Nothing}=nothing,
+                              subdir::AbstractString="",
+                              )
     version_info = Dict{String, Any}("git-tree-sha1" => string(tree_hash))
+    if !isnothing(commit_hash)
+        version_info["git-commit-sha1"] = commit_hash
+    end
+    if !isnothing(tag_hash)
+        version_info["git-tag-sha1"] = tag_hash
+    end
+    if !isnothing(tag_name)
+        version_info["git-tag-name"] = tag_name
+    end
+    if subdir != ""
+        version_info["git-tree-path"] = subdir
```

Comment:
Should this be `subdir`? See https://github.com/JuliaLang/Pkg.jl/issues/3718#issuecomment-1873038824

Reply:
Renamed to be `"subdir"`